### PR TITLE
crypto/move_block_parsing_logic_to_bom_cryptocurrency

### PR DESCRIPTION
**Pull request by Binary-Mohanad**
PR changes:
- Remove the subscription for the confirmed transactions and the related logic
- Add subscription for the new blocks

### CODE REVIEW

File: lib/Net/Async/Blockchain/BTC.pm [hunk 1]
```diff
@@ -145,168 +88,15 @@ L<Ryu::Source>
 
 =cut
 
-sub subscribe {
+async sub subscribe {
```
Comment (tahir-deriv) [inline lib/Net/Async/Blockchain/BTC.pm:91]: question : why you need to make this sub async?

File: lib/Net/Async/Blockchain/BTC.pm [hunk 2]
```diff
@@ -145,168 +88,15 @@ L<Ryu::Source>
...
-        push(@transactions, $transaction);
-    }
-
-    for my $transaction (@transactions) {
-        $self->source->emit($transaction);
-    }
-
-    return 1;
+    my $subscription_event = $subscription_dictionary{$subscription} or die "Invalid or not implemented subscription";
+    return $self->zmq_client->subscribe($subscription_event)->map(
```
Comment (tahir-deriv) [inline lib/Net/Async/Blockchain/BTC.pm:94]: question : could you please help me understand the use of `map` here?

File: lib/Net/Async/Blockchain/BTC.pm (hunk 1 above)
Comment (Binary-Mohanad) [reply lib/Net/Async/Blockchain/BTC.pm:91]: it's a shared function we overwrite it here, so it's async version in other packages like Etehreum

File: lib/Net/Async/Blockchain/BTC.pm (hunk 2 above)
Comment (Binary-Mohanad) [reply lib/Net/Async/Blockchain/BTC.pm:94]: The [Ryu source map](https://metacpan.org/pod/Ryu::Source#map) is a bit similar to the origin Perl map function [(Perl map document)](https://perldoc.perl.org/functions/map)
it performs some operations on the item then it returns the result, in our case, we formate the message then we return a formatted message.

**Pull request merged**